src/Translator.php

```diff
@@ -55,22 +55,19 @@ public function run()
         $this->outputFormatter = new OutputFormatter(true);
         $this->checkEnv();
         $this->setModulePaths();
+        $this->log('Updating translations for ' . count($this->modulePaths) . ' module(s)');
         if ($this->doTransifexPullAndUpdate) {
-            $this->log('Updating translations for ' . count($this->modulePaths) . ' module(s)');
-            $this->storeJson();
-            $this->storeYaml();
+            $this->log('Starting transifex pull block');
+            $this->backupContent();
             $this->setJsonAndYmlFileTimes();
             $this->transifexPullSource();
-            $this->mergeYaml();
-            $this->removeEnglishStringsFromYamlTranslations();
-            $this->cleanYaml();
-            $this->mergeJson();
-            $this->removeEnglishStringsFromJsonTranslations();
-            $this->removeEmptyYamlFiles();
-            $this->removeEmptyJsFiles();
+            $this->mergeAndCleanContent();
         }
         if ($this->doCollectStrings) {
```

Comment:
We probably need an additional check here to ensure that `$this->doTransifexPullAndUpdate() && $this->doCollectStrings()` are not both true at the same time, otherwise I'd say things will end up in a pretty bad state.  Simply throwing an exception is fine

Reply:
Discussed offline, may do this as separate issue